File: Jaar 2 FHICT/React Individueel project/src/services/PostService.js
import axios from "axios";

const POST_API_BASE_URL = "http://localhost:8080/api/v1/Post"
class PostService{

    getposts(forumID){
        return axios.get(POST_API_BASE_URL+"s",forumID)
    }
    getpost(postID){
        return axios.get(POST_API_BASE_URL,postID)
    }
    postUsers(Post){

        return axios.post(POST_API_BASE_URL,Post)
    }
    putUsers(Post){
        return axios.put(POST_API_BASE_URL,Post)
    }
    deleteUser(PostID){
        return axios.delete(POST_API_BASE_URL,PostID)
    }
}

export default new PostService()

File: Jaar 2 FHICT/Java Springboot Individueel project/src/main/java/com/fhict/sociable/Service/PostService.java
package com.fhict.sociable.Service;

import com.fhict.sociable.DTO.PostDTO;
import com.fhict.sociable.DTO.PostDTOConverter;
import com.fhict.sociable.Exceptions.ResourceNotFoundException;
import com.fhict.sociable.Model.Forum;
import com.fhict.sociable.Model.Post;
import com.fhict.sociable.Repository.PostRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class PostService {

    @Autowired
    PostRepository postRepository;

    @Autowired
    PostDTOConverter converter;
    public List<Post> getAllPosts(){

        return postRepository.findAll();
    }
    public List<Long> getPostByForum(long forumId){
        return postRepository.findPostsByForumId(forumId);
    };

    public Optional<Post> getPost(long postId){
        return postRepository.findById(postId);
    }

    public Post savePost(PostDTO post){
        return postRepository.save(converter.DtoToObject(post));
    }

    public ResponseEntity<Post> updatePost(Post postdetails)throws ResourceNotFoundException{
        Post post = postRepository.findById(postdetails.getId()).orElseThrow(() -> new ResourceNotFoundException("Post not found for this id :: " + postdetails.getId()));

        post.setPostTitle(postdetails.getPostTitle());
        post.setDislikes(post.getDislikes());
        post.setLikes(post.getLikes());
        post.setPostLink(post.getPostLink());

        final Post Updatedpost =postRepository.save(post);

        return ResponseEntity.ok(Updatedpost);
    }

    public boolean deletePost(long PostID)throws ResourceNotFoundException{
        Post post = postRepository.findById(PostID).orElseThrow(() -> new ResourceNotFoundException("Post not found for this ID :: "+ PostID));

        postRepository.delete(post);
        return true;
    }
}
